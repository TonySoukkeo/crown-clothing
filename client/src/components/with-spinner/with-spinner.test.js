import React from "react";
import { configure, shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

import WithSpinner from "./with-spinner.component";
import Spinner from "../spinner/spinner.component";

configure({
  adapter: new Adapter(),
});

describe("WithSpinner HOC", () => {
  let wrapper;
  const TestComponent = () => <div />;

  beforeEach(() => {
    const WrappedComponent = WithSpinner(TestComponent);

    wrapper = shallow(<WrappedComponent isLoading={true} />);
  });

  it("renders withSpinner without crashing", () => {
    expect.assertions(1);
    expect(wrapper).toMatchSnapshot();
  });

  describe("if loading is true", () => {
    it("should render Spinner component", () => {
      expect.assertions(1);
      expect(wrapper.find(Spinner).exists()).toEqual(true);
    });

    it("should not render wrapped copmonent", () => {
      expect.assertions(1);
      expect(wrapper.find(TestComponent).exists()).toEqual(false);
    });
  });

  describe("if loading is false", () => {
    it("should not render spinner component", () => {
      const WrappedComponent = WithSpinner(TestComponent);

      const wrapper2 = shallow(<WrappedComponent isLoading={false} />);

      expect.assertions(1);
      expect(wrapper2.find(TestComponent).exists()).toEqual(true);
    });

    it("should not render spinner", () => {
      const WrappedComponent = WithSpinner(TestComponent);

      const wrapper2 = shallow(<WrappedComponent isLoading={false} />);

      expect.assertions(1);
      expect(wrapper2.find(Spinner).exists()).toEqual(false);
    });
  });
});
