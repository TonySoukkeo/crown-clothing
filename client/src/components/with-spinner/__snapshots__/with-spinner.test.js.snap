// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WithSpinner HOC renders withSpinner without crashing 1`] = `ShallowWrapper {}`;
